### ISSUE TITLE
Two Stage Fallback not working as described

### ISSUE THREAD

**Issue by AllBecomesGood** [RasaHQ/rasa#4123]
**Rasa version**: 1.1.8

**Rasa X version** (if used & relevant):

**Python version**: 3.7.3

**Operating system** (windows, osx, ...): win 10

**Issue**:
https://github.com/RasaHQ/rasa/blob/master/rasa/core/policies/two_stage_fallback.py

It simply does not ask the user to rephrase as promised.
```class TwoStageFallbackPolicy(FallbackPolicy):
    """ This policy handles low NLU confidence in multiple stages.
        If a NLU prediction has a low confidence score,
        the user is asked to affirm whether they really had this intent.
        If they affirm, the story continues as if the intent was classified
        with high confidence from the beginning.
      **If they deny, the user is asked to rephrase his intent**.
        [...]"""

```
**Command or request that led to error**:
rasa shell
```
Your input ->  hi
Hey! How are you?
Your input ->  yes no kinda sad really mhm
? Did you mean 'user_feels_sad'?  2: No (/out_of_scope)
? Did you mean 'user_feels_sad'?  2: No (/out_of_scope)
utter default: my sentence in domain.yml blabla
```

Now obviously I should say yes, but I want it to ask me to rephrase.

**Content of configuration file (config.yml)** (if relevant):
```policies:
  - name: MemoizationPolicy
  - name: KerasPolicy
    epochs: 125
  - name: MappingPolicy
  - name: TwoStageFallbackPolicy
    nlu_threshold: 0.9
    core_threshold: 0.9

and same when I change it to:

  - name: TwoStageFallbackPolicy
    nlu_threshold: 0.9
    core_threshold: 0.9
    fallback_core_action_name: "action_default_fallback"
    fallback_nlu_action_name: "action_default_fallback"
    deny_suggestion_intent_name: "out_of_scope"

```
I set it to 0.9 because my test sentence achieves 0.74 and the highest is the correct result (*user_feels_sad). Otherwise it asks me twice to confirm a wrong intent.

**Content of domain file (domain.yml)** (if relevant):
```yml
intents:
  - greet
  - goodbye
  - affirm
  - deny
  - mood_great
  - user_feels_sad
  - out_of_scope

actions:
- utter_greet
- utter_cheer_up
- utter_did_that_help
- utter_happy
- utter_goodbye
- utter_default

templates:
  utter_greet:
  - text: "Hey! How are you?"

  utter_cheer_up:
  - text: "Here is something to cheer you up:"
    image: "https://i.imgur.com/nGF1K8f.jpg"

  utter_did_that_help:
  - text: "Did that help you?"

  utter_happy:
  - text: "Great carry on!"

  utter_goodbye:
  - text: "Bye"
  
  action_default_fallback:
  - text: "text for action_default_fallback"
  
  out_of_scope:
  - text: "out_of_scope utterance template blablabla"
  
  utter_restart:
  - text: "Restarted the bot!!"
  
  utter_default:
  - text: "utter default: my sentence in domain.yml blabla"
```

So I might have set this up not 100% correctly, but I did according to the tutorial. So maybe the tutorial is incomplete, or it is a bug. Either way I can't figure out how to fix it, as the demo bot for example uses a bunch of custom actions whereas I want the default implementation in this case

**Comment by msamogh** [comment]
Thanks for raising this issue, @federicotdn will get back to you about it soon.

**Comment by AllBecomesGood** [comment]
I found how to get it to ask for the rephrase after running rasa shell --debug which showed that rephrase ran, but returned [] nothing. In domain.yml I had to add:
```
templates:
  utter_ask_rephrase:
  - text: "Could you rephrase this please blablabla?"
```
And then in rasa shell testing the bot it worked:
```
Your input ->  hi
Hey! How are you?
Your input ->  i am kind of sad really if you must know
? Did you mean 'user_feels_sad'?  2: No (/out_of_scope)
Could you rephrase this pls?
Your input ->  i am kind of sad really if you must knowwwww
? Did you mean 'user_feels_sad'?  2: No (/out_of_scope)
utter default: my sentence in domain.yml blabla

Your input ->  i am sad
utter default: my sentence in domain.yml blabla
Your input ->  i am kind of sad really if you must knoww
? Did you mean 'user_feels_sad'?  1: Yes (/user_feels_sad)
utter default: my sentence in domain.yml blabla
```

As you can see it works (it breaks because there's hardly any story data i think, so it just doesn't know what to do when you tell it you're sad several times in a row).

I suggest that this would be added to the tutorial / and or the docs. One strange thing is that it apparently didn't matter whether I added the action in domain.yml under actions: as it worked either way with no errors.
```
actions:
- [... buncha stuff...]
- utter_ask_rephrase
```

**Comment by AllBecomesGood** [comment]
Ok i'm an idiot it is in the docs
